forbidigo/patterns_test.go

```diff
@@ -0,0 +1,26 @@
+package forbidigo
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseValidPattern(t *testing.T) {
+	ptrn, err := parse(`fmt\.Errorf`)
+	require.Nil(t, err)
+	assert.Equal(t, `fmt\.Errorf`, ptrn.pattern.String())
+}
+
+func TestParseValidPatternWithCustomMessage(t *testing.T) {
```

Comment:
If we want to be thorough here we could also add a test for a nested subgroup, ensuring the recursion works.

I personally also like the structure of having a single `TestParse` method at the top-level and writing a `t.Run()` for each subtest. A table-style test might also be appropriate here, although there aren't that many variants.